2022/incoming/catching-dreams.pod

```diff
@@ -0,0 +1,118 @@
+Title: Catching dreams
+Topic: Mango
+
+=encoding utf8
+
+=head1 Catching dreams
+
+=head2 Listening to dreams
+
+During the year, some kids are writing "letters to Santa" or just sharing loudly what they want to "order" for Christmas, but it's not always the case...
+Sometimes kids are only I<dreaming> what they want.
+
+Normally, dreaming is not the most effective way to "order" a present for Christmas... But Santa thought about everything and there is a special room in North Pole complex for this purpose. It's an all white, circle room with chairs arranged in the middle. Elves arrive at the room when it is night time in their assigned timezone.
+
+They bring a pen and a notebook, and they listen to childrens' dreams and take notes.
+
+=head2 Gathering notebooks
+
+Elves capture notes on childrens' dreams all year long and, hence, approaching Christmas there are plenty of notebooks to put together and add to other wishes.
+
+One of the elves had the idea to use L<MongoDB|https://www.mongodb.com/> to store data.
+
+Santa complained MongoDB was not Perl friendly (e.g. struggles with drivers...) which had in its history taken a controversial license... But the elves are not disciplined and decided to go with it anyway.
+
+1. First they installed MongoDB:
+
+	$ sudo apt-get install mongodb
+
+2. Then they started a shell:
+
+	$ mongo
+	MongoDB shell version v3.6.8
+	connecting to: mongodb://127.0.0.1:27017
+	MongoDB server version: 5.0.9
+	>
+
+3. The elves created a new Database:
+
+	> use christmas2022
+	switched to db christmas2022
+
+=head3 Import dreams
+
+Despite being undisciplined, the elves followed a convention in their notes and all of their notebooks are all well-formatted as CSV. 
+
+At the end of the night, a notebook looks like this:
+
+	what,who,where
+	bicycle,Tom,Atlanta
+	Barbie,Lisa,Berlin
+	Transformer,Billy,Houston
+	ball,Ismail,Atlanta
+
+It's easy to import these notes into MongoDB:
+
+	$ mongoimport -d christmas2022 -c gifts --type csv --file gifts.csv --headerline
+	2022-08-04T09:57:42.400+0200	connected to: mongodb://localhost/
+	2022-08-04T09:57:42.408+0200	4 document(s) imported successfully. 0 document(s) failed to import.
+
+And you can quickly verify that it is imported:
+
+	> db.gifts.find()
+	{ "_id" : ObjectId("62eb96777d9a1bc25a68b25c"), "what" : "Transformer", "who" : "Billy", "where" : "Houston" }
+	{ "_id" : ObjectId("62eb96777d9a1bc25a68b25d"), "what" : "bicycle", "who" : "Tom", "where" : "Atlanta" }
+	{ "_id" : ObjectId("62eb96777d9a1bc25a68b25e"), "what" : "ball", "who" : "Ismail", "where" : "Atlanta" }
+	{ "_id" : ObjectId("62eb96777d9a1bc25a68b25f"), "what" : "Barbie", "who" : "Lisa", "where" : "Berlin" }
+
+=head2 Using Perl to access dreams
+
+To integrate this database in the North Pole code, the elves need to interact with it in Perl.
+
+Let's install L<Mango>:
+
+	$ sudo cpanm Mango
+
+Then they can access records with Perl methods:
+
+=begin perl
+	
+	use Mango;
+	use feature state;
```

Comment:
```suggestion
	use feature qw( state );
```

Looks like this is copy/paste from a broken SYNOPSIS: https://github.com/oliwer/mango/commit/260788cb5e3ac1c0f22f449eda61d1b941ec577f